test/e2e/user_management_test.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/stretchr/testify/suite"
+	apierros "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUserManagement(t *testing.T) {
+	suite.Run(t, &userManagementTestSuite{})
+}
+
+type userManagementTestSuite struct {
+	suite.Suite
+	namespace   string
+	testCtx     *framework.TestCtx
+	awaitility  *wait.Awaitility
+	hostAwait   *wait.HostAwaitility
+	memberAwait *wait.MemberAwaitility
+}
+
+func (s *userManagementTestSuite) SetupSuite() {
+	userSignupList := &v1alpha1.UserSignupList{}
+	s.testCtx, s.awaitility = testsupport.WaitForDeployments(s.T(), userSignupList)
+	s.hostAwait = s.awaitility.Host()
+	s.memberAwait = s.awaitility.Member()
+	s.namespace = s.awaitility.HostNs
+}
+
+func (s *userManagementTestSuite) TearDownTest() {
+	s.testCtx.Cleanup()
+}
+
+func (s *userManagementTestSuite) TestUserDisabled() {
+	// Create UserSignup
+	userSignup := createAndApproveSignup(s.T(), s.awaitility, "janedoe")
+
+	// Get MasterUserRecord
+	mur, err := s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username)
+	require.NoError(s.T(), err)
+
+	// Get the UserAccount
+	userAccount, err := s.memberAwait.WaitForUserAccount(mur.Name,
+		wait.UntilUserAccountHasConditions(provisioned()),
+		wait.UntilUserAccountMatchesMur(mur.Spec, mur.Spec.UserAccounts[0].Spec))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userAccount)
+
+	// User should be created
+	_, err = s.memberAwait.WaitForUser(userAccount.Name)
+	require.NoError(s.T(), err)
+
+	// Identity should be created
+	_, err = s.memberAwait.WaitForIdentity(toIdentityName(userAccount.Spec.UserID))
+	require.NoError(s.T(), err)
+
+	// Get MasterUserRecord
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username)
+	require.NoError(s.T(), err)
+
+	// Disable MUR
+	mur.Spec.Disabled = true
+	err = s.awaitility.Host().Client.Update(context.TODO(), mur)
+	require.NoError(s.T(), err)
+
+	// Wait until the UserAccount status is disabled
+	userAccount, err = s.memberAwait.WaitForUserAccount(mur.Name,
+		wait.UntilUserAccountHasConditions(disabled()))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userAccount)
+
+	// Wait until the MUR status is disabled
+	mur, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Spec.Username,
+		wait.UntilMasterUserRecordHasConditions(disabled()))
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), mur)
+
+	// Check that the UserAccount is now set to disabled
+	require.True(s.T(), userAccount.Spec.Disabled)
+
+	// Check the User is deleted
+	user := &userv1.User{}
+	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Name: userAccount.Namespace}, user)
+	require.Error(s.T(), err)
+	assert.True(s.T(), apierros.IsNotFound(err))
+
+	// Check the Identity is deleted
+	identity := &userv1.Identity{}
+	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Name: toIdentityName(userAccount.Spec.UserID)}, identity)
+	require.Error(s.T(), err)
+	assert.True(s.T(), apierros.IsNotFound(err))
```

Comment:
@MatousJobanek i don't think we need to wait in this circumstance because we actually wait for the condition disabled. If the condition disabled has been set then for sure the user/identity does not exist as this must be done prior to the condition being set - unless i'm missing something?: https://github.com/codeready-toolchain/toolchain-e2e/pull/75/files/56e82024949e457be7974785e6245457b70857d1#diff-4f14cf49a084497bcfac1b29c3bbf79bR79

Reply:
It should have the same result just less code, but I got your point so keep it as it is.

Reply:
The difference between these two approaches is that if we always wait for User/Identity (vs. using a single GET) then the test will always pass even if we set Disabled status before the User/Identity actually deleted.

Reply:
Good point @alexeykazakov